static/app/utils/fields/index.ts

```diff
@@ -118,6 +119,7 @@ export enum FieldKey {
   USER_ID = 'user.id',
   USER_IP = 'user.ip',
   USER_USERNAME = 'user.username',
+  USER_SEGMENT = 'user.segment',
```

Comment:
`user.segment` should be fine to add, we should follow up and also add this to all other places.